weatherman.py

```diff
@@ -0,0 +1,189 @@
+import os
+import sys
+import shutil
+import argparse
+import datetime as dt
+
+
+from report_printer import year_report, month_report, month_chart
+from file_handler import extract_files, handle_csv, handle_xlsx
+from file_handler import delete_files
+from assist_functions import date_to_int
+
+# globals
+# main data structure to store weather dataset
+weather_dataset = {}
+
+
+def handle_sys_argv():
+    """Validates and parses sys arguments
+    """
+
+    parser = argparse.ArgumentParser()
+    parser.add_argument(
+        'path', type=check_dir_path,
+        help="path to folder containing weatherfiles.zip")
+    parser.add_argument(
+        '-e', nargs='?', type=check_year,
+        help="enter year to get yearly report")
+    parser.add_argument(
+        '-a', nargs='?', type=check_month_year,
+        help="enter year/month to get monthly report")
+    parser.add_argument(
+        '-c', nargs='?', type=check_month_year,
+        help="enter year/month to get monthly chart")
+    args = parser.parse_args()
+
+    if args.e == args.a == args.c == None:
+        print("No arguments given")
+        print_correct_format()
+        exit()
+
+    return args
+
+
+def check_dir_path(path):
+    """ path check for argparse
+    """
+
+    if os.path.isdir(path):
+        return path
+    else:
+        raise argparse.ArgumentTypeError(
+            f"readable_dir:{path} is not a valid path")
+
+
+def check_month_year(date):
+    """ date check for argparse
+    """
+
+    try:
+        date = dt.datetime.strptime(date, "%Y/%m")
+        return date
+    except ValueError:
+        raise argparse.ArgumentTypeError(
+            f"Date: {date} is not a valid date.\n"
+            "Correct format e.g. 2012/6")
+
+
+def check_year(date):
+    """ year check for argparse
+    """
+
+    try:
+        date = dt.datetime.strptime(date, "%Y")
+        return date
+    except ValueError:
+        raise argparse.ArgumentTypeError(
+            f"Year: {date} is not a valid date.\n" +
+            "Correct year e.g. 2011")
+
+
+def print_correct_format():
+    """Prints examples of correct command format and exits code
+
+    Helper Function that assists, when user enters invalid command
+    to execute code
+    """
+
+    print("Following are the correct command examples:\n")
+    print("python3 weatherman.py /path-to-zipfile -e 2011 ")
+    print("python3 weatherman.py /path-to-zipfile -e 2016 -a 2007/6 -c 2009/5\n")
+    exit()
+
+
+def date_exists(year, month=-1):
+    """Returns True if given month/year exists in weather dataset
+    """
+
+    if year in weather_dataset:
+        if month != -1:
+            if month in weather_dataset[year]:
+                return True
+            else:
+                return False
+        return True
+    else:
+        return False
```

Comment:
@ahmadx16 Check how we can simplify these lots of if-else conditions
```suggestion
def date_exists(year, month=None):
    """Returns True if given month/year exists in weather dataset
    """
    if year not in weather_dataset:
        return False
    if month != -1:
        return month in weather_dataset[year]
    return True        
```

Another good approach is to use regex to do such type of matching, that's also efficient in such scenarios.